pkg/graveler/graveler.go

```diff
@@ -1217,11 +1229,28 @@ func (g *Graveler) UpdateBranch(ctx context.Context, repository *RepositoryRecor
 	return newBranch, nil
 }
 
-// prepareForCommitIDUpdate will check that the staging area is empty,
-// before adding a new staging token. It is best to use it before changing
-// the branch HEAD as a preparation for deleting the staging area.
-// see issue #3771 for more information on the algorithm used.
-func (g *Graveler) prepareForCommitIDUpdate(ctx context.Context, repository *RepositoryRecord, branchID BranchID) error {
+// monitorRetries reports the number of retries of operation while updating
+// branchID of repository.
+func (g *Graveler) monitorRetries(ctx context.Context, retries int, repositoryID RepositoryID, branchID BranchID, operation string) {
+	kvRetriesCounter.WithLabelValues(operation, string(repositoryID)).Add(float64(retries))
+	l := g.log(ctx).WithFields(logging.Fields{
+		"retries":    retries,
+		"repository": repositoryID,
+		"branch":     branchID,
+		"operation":  operation,
+	})
+	if retries > 0 {
+		l.Info("Operation retried")
+	} else {
+		l.Trace("No retries needed")
+	}
```

Comment:
Can we have just one place where we log retries? You have Info + Trace here and Debug below. I think `Info` is too verbose for a somewhat expected retry, but anyway let's have it there or here and not both.